check/src/unify_type.rs

```diff
@@ -133,11 +133,8 @@ impl<I> Substitutable for ArcType<I> {
     type Variable = TypeVariable;
 
     fn new(id: u32) -> ArcType<I> {
-        Type::variable(TypeVariable::new(id))
-    }
-
-    fn from_variable(var: TypeVariable) -> ArcType<I> {
-        Type::variable(var)
+        // FIXME: using `Kind::hole()` here results in test failures
+        Type::variable(TypeVariable::new(id, Kind::typ()))
```

Comment:
The `Kind::Hole` is probably never replaced with an actual type. Maybe check where this function is called? It is possible that `Kind:Type` is either correct in all cases or the method could even be removed so that only the `from_variable` function is called (which receives the kind from the caller).